### PR TITLE
fix(conversation): fix bug open last chating conversation

### PR DESCRIPTION
**Pull request by developerfred**
### What does this do?
When visiting zos without a conversationId specified in the URL, the most recent conversation should open

#### Things to consider:
  1. How will this affect security?
  1. How will this affect performance?
  1. Does this change any APIs?

### CODE REVIEW

File: src/components/messenger/list/conversation-list-panel/index.tsx [hunk 1]
```diff
@@ -53,6 +53,30 @@ export class ConversationListPanel extends React.Component<Properties, State> {
     this.scrollContainerRef = React.createRef();
   }
 
+  componentDidMount() {
+    this.openMostRecentConversation();
+  }
+
+  openMostRecentConversation = () => {
```
Comment (domw30) [inline src/components/messenger/list/conversation-list-panel/index.tsx:60]: This won't be required, we shouldn't be handling this sort of functionality at component level. Please take a look at the saga where the logic lives for handling conversations. We already have a `openFirstConversation` saga function that handles the opening of the first conversation. If there is an issue, this is where you need to investigate. It would probably be worth looking into the room events, if they are being handled/received correctly to determine what conversation to open.

`src/store/channels/saga.ts` 
`src/lib/chat/matrix-client.ts`

File: src/components/messenger/list/conversation-list-panel/conversation-list-panel.scss [hunk 2]
```diff
@@ -58,7 +58,7 @@ $side-padding: 16px;
       }
 
       .messages-list__tab-badge {
-        display: none;
+        display: block;
```
Comment (domw30) [inline src/components/messenger/list/conversation-list-panel/conversation-list-panel.scss:61]: Is this change related to opening the last conversation bug?

Looks like this change is to fix a different issue, the unread count on favourites tab? If so, this should be fixed in a different pull request/commit. Try to keep commits separate

**Review by domw30 — CHANGES_REQUESTED** [review]

File: src/components/messenger/list/conversation-list-panel/conversation-list-panel.scss (hunk 2 above)
Comment (developerfred) [reply src/components/messenger/list/conversation-list-panel/conversation-list-panel.scss:61]: ok, I will do this!

File: src/store/channels/saga.ts [hunk 3]
```diff
@@ -142,6 +146,7 @@ export function* roomUnfavorited(action) {
 }
 
 export function* saga() {
+  yield call(openFirstConversation);
```
Comment (ratik21) [inline src/store/channels/saga.ts:149]: @developerfred this won't make this call upon first load. In order to do that, you should probably look into the page-load saga.

**Review by ratik21 — COMMENTED** [review]
@developerfred @domw30 do we need this? 

I tested on prod & local, if i open `localhost:3000` directly, then it still takes me to the first conversation in the list. What's the issue we're trying to solve w/ this PR?

image

this ^ already happens on prod.

File: src/store/channels/saga.ts (hunk 3 above)
Comment (developerfred) [reply src/store/channels/saga.ts:149]: ok, cool!

**Comment by developerfred** [comment]
> @developerfred @domw30 do we need this?
> 
> I tested on prod & local, if i open `localhost:3000` directly, then it still takes me to the first conversation in the list. What's the issue we're trying to solve w/ this PR?
> 
> image
> this ^ already happens on prod.

**To reproduce the error you need the following flow**: 

- start a chat with a new user, write some messages
- then open a new user, write one again

agpra opens the chat in a new tab and checks who will appear first, currently chat A appears and not chat B, which was the last one you talked to. This pull request fixes this!

**Comment by ratik21** [comment]
@developerfred can paste a video here reproducing this? Seems to work fine on mine. 

This is how it looks 

https://github.com/zer0-os/zOS/assets/33264364/c1ec6e21-557f-49cd-b8d7-613b071a2625

This is the correct (expected) behaviour.
